Compilation in MSYS2/MinGW with GCC 13.1 fails: symbols not declared in this scope

### Godot version

4.0

### System information

Windows 10, NVidiaRTX 2070

### Issue description

When trying to compile using MinGW with GCC 13.1 compilation fails throwing a bunch of errors like this one:

```
G:/MABS/build/glslang-git/glslang/MachineIndependent/SymbolTable.h:71:1: note: 'uint32_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?

G:/MABS/build/glslang-git/glslang/MachineIndependent/SymbolTable.h:662:22: error: 'uint64_t' does not name a type
  662 |     static constexpr uint64_t uniqueIdMask = (1LL << LevelFlagBitOffset) - 1;
```

It looks like the problem comes from third parties: vhacd, openxr and glslang

I've solved the issue by adding:
**#include <stdint.h>**

to 
**\thirdparty\vhacd\inc\vhacdManifoldMesh.h
\thirdparty\openxr\src\common\platform_utils.hpp
\thirdparty\glslang\glslang\Include\Common.h**

I'm not sure how to proceed after finding the fix...

### Steps to reproduce

Try compiling the editor from Master or version 4.0 using MinGW with GCC 13.1

### Minimal reproduction project

N/A

--- a/thirdparty/glslang/glslang/Include/Common.h
+++ b/thirdparty/glslang/glslang/Include/Common.h
@@ -45,6 +45,7 @@
 #include <cmath>
 #endif
 #include <cstdio>
+#include <cstdint>
 #include <cstdlib>
 #include <list>
 #include <map>

--- a/thirdparty/openxr/patches/fix-gcc13-stdint.patch
+++ b/thirdparty/openxr/patches/fix-gcc13-stdint.patch
@@ -1,0 +1,12 @@
+diff --git a/thirdparty/openxr/src/common/platform_utils.hpp b/thirdparty/openxr/src/common/platform_utils.hpp
+index 85d5cdab10..2d870cfea7 100644
+--- a/thirdparty/openxr/src/common/platform_utils.hpp
++++ b/thirdparty/openxr/src/common/platform_utils.hpp
+@@ -11,6 +11,7 @@
+ 
+ #include "xr_dependencies.h"
+ #include <string>
++#include <stdint.h>
+ #include <stdlib.h>
+ 
+ // OpenXR paths and registry key locations

--- a/thirdparty/openxr/src/common/platform_utils.hpp
+++ b/thirdparty/openxr/src/common/platform_utils.hpp
@@ -11,6 +11,7 @@
 
 #include "xr_dependencies.h"
 #include <string>
+#include <stdint.h>
 #include <stdlib.h>
 
 // OpenXR paths and registry key locations

--- a/thirdparty/vhacd/0006-fix-gcc13.patch
+++ b/thirdparty/vhacd/0006-fix-gcc13.patch
@@ -13,3 +13,26 @@ index 132bdcfb3e..925584cf52 100644
  namespace VHACD {
  //!    Incremental Convex Hull algorithm (cf. http://cs.smith.edu/~orourke/books/ftp.html ).
  enum ICHullError {
+diff --git a/thirdparty/vhacd/inc/vhacdManifoldMesh.h b/thirdparty/vhacd/inc/vhacdManifoldMesh.h
+index a48f53c5c5..5eed4e13aa 100644
+--- a/thirdparty/vhacd/inc/vhacdManifoldMesh.h
++++ b/thirdparty/vhacd/inc/vhacdManifoldMesh.h
+@@ -18,6 +18,11 @@ All rights reserved.
+ #include "vhacdCircularList.h"
+ #include "vhacdSArray.h"
+ #include "vhacdVector.h"
++
++// -- GODOT start --
++#include <cstdint>
++// -- GODOT end --
++
+ namespace VHACD {
+ class TMMTriangle;
+ class TMMEdge;
+@@ -139,4 +144,4 @@ private:
+     friend class ICHull;
+ };
+ }
+-#endif // VHACD_MANIFOLD_MESH_H
+\ No newline at end of file
++#endif // VHACD_MANIFOLD_MESH_H

--- a/thirdparty/vhacd/inc/vhacdManifoldMesh.h
+++ b/thirdparty/vhacd/inc/vhacdManifoldMesh.h
@@ -18,6 +18,11 @@ All rights reserved.
 #include "vhacdCircularList.h"
 #include "vhacdSArray.h"
 #include "vhacdVector.h"
+
+// -- GODOT start --
+#include <cstdint>
+// -- GODOT end --
+
 namespace VHACD {
 class TMMTriangle;
 class TMMEdge;

